packages/checkout/widgets-lib/src/widgets/connect/views/SwitchNetworkZkEVM.tsx

```diff
@@ -38,7 +39,63 @@ export function SwitchNetworkZkEVM() {
       controlType: 'Button',
     });
 
+    if (!provider.provider.request) return;
+
+    const currentChainId = provider.provider.request({ method: 'eth_chainId', params: [] });
+    // eslint-disable-next-line radix
+    const parsedChainId = parseInt(currentChainId.toString());
+
+    if (parsedChainId === getL2ChainId(checkout.config)) {
+      connectDispatch({
+        payload: {
+          type: ConnectActions.SET_PROVIDER,
+          provider,
+        },
+      });
+
+      viewDispatch({
+        payload: {
+          type: ViewActions.UPDATE_VIEW,
+          view: {
+            type: ConnectWidgetViews.SUCCESS,
+          },
+        },
+      });
+
+      return;
+    }
+
+    const walletName = (provider.provider as any)?.session?.peer?.metadata?.name.toLowerCase();
+
     try {
+      if (isWalletConnectProvider(provider) && walletName === 'metamask') {
+        try {
```

Comment:
The 2 layers of try catch feels a little redundant.. might be worth flattening them out to different code blocks?

Reply:
Just want to make sure that the flow is still correct for WalletConnect if flattening this out. For example we're catching the first error and skipping then going to switch

Reply:
Yeah, not sure if it'll work as we want if we flatten this out. Happy to pair to see how we could flatten if there's a way.

Reply:
All good!